### PR TITLE
Remove the linkerd2-task crate

### PR DESCRIPTION
**Pull request by olix0r**
Most of the logic in `linkerd2-task` is not actually used.

This change moves the used part of this crate to `task-test-util`, and
removes the rest of the linkerd-specific task infrastructure in favor of
using tokio directly.

### CODE REVIEW

File: linkerd/app/src/lib.rs [hunk 1]
```diff
@@ -155,28 +149,24 @@ where
...
+
+                    proxy_parts.build_proxy_task(drain_rx);
+                    trace!("main task spawned");
+
+                    shutdown_signal.and_then(move |()| {
+                        debug!("shutdown signaled");
+                        drain_tx.drain().map(|()| debug!("shutdown complete"))
+                    })
+                })
+                .instrument(info_span!("main")),
```
Comment (olix0r) [inline linkerd/app/src/lib.rs]: shouldn't do this...

Comment (hawkw) [reply linkerd/app/src/lib.rs]: ?

Comment (olix0r) [reply linkerd/app/src/lib.rs]: i don't think i actually want `main:` prefixed on all of the log messages

File: linkerd/task-test-util/Cargo.toml [hunk 2]
```diff
@@ -1,17 +1,10 @@
 [package]
-name = "linkerd2-task"
+name = "linkerd2-task-test-util"
```
Comment (hawkw) [inline linkerd/task-test-util/Cargo.toml]: WDYT about just making this a general `test-util` crate? We might want to put some other stuff that's currently in the integration test crate in there as well, in the future, if it's used in unit tests as well...

File: linkerd/app/src/lib.rs (hunk 1 above)
Comment (hawkw) [reply linkerd/app/src/lib.rs]: ah, yeah. probably get rid of this...

**Pull request merged**